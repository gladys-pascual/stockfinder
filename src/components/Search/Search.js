import "./Search.scss";

const Search = () => {
  return (
    <form className="search-form">
      <div className="form-wrapper">
        <input
          className="search-input"
          type="text"
          name="query"
          placeholder="Search for symbol or company name..."
        ></input>
        <button type="submit" id="submit">
          <p>Search</p>
        </button>
      </div>
    </form>
  );
};

export default Search;
