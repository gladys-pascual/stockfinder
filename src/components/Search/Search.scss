@import "../../assets/variables.scss";

.search-form {
  display: flex;
  justify-content: center;
  padding: 50px 0px;

  .form-wrapper {
    border: 1px solid lightgrey;
    border-radius: 30px;
    padding-left: 20px;
    @media screen and (max-width: 500px) {
      padding-left: 12px;
    }
  }
  .search-input {
    width: 50vw;
    letter-spacing: 1px;
    border: none;
    font-size: 1em;
    @media screen and (max-width: 780px) {
      width: 70vw;
    }
    @media screen and (max-width: 500px) {
      font-size: 12px;
      width: 65vw;
    }
  }
  button {
    width: 120px;
    background-color: $theme-color;
    border-radius: 30px;
    border: none;
    height: 46px;
    @media screen and (max-width: 780px) {
      width: 80px;
    }
    p {
      text-transform: uppercase;
      letter-spacing: 1px;
      font-weight: 600;
      margin: 0px;
      color: white;
      @media screen and (max-width: 780px) {
        font-size: 12px;
      }
    }
  }
  button:hover {
    background-color: $theme-color + 20%;
  }
}
