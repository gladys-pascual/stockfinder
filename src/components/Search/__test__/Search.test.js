import { render } from "@testing-library/react";
import Search from "../Search";

test("render Search component as expected", () => {
  const { container } = render(<Search />);

  expect(container).toMatchSnapshot();
});
