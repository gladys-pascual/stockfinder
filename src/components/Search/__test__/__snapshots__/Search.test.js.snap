// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Search component as expected 1`] = `
<div>
  <form
    class="search-form"
  >
    <div
      class="form-wrapper"
    >
      <input
        class="search-input"
        name="query"
        placeholder="Search for symbol or company name..."
        type="text"
      />
    </div>
  </form>
  <div
    class="query-results-list"
  >
    <ul
      class="results"
    />
  </div>
</div>
`;
